### PR TITLE
Set `-fno-ignore-asserts` in GHA

### PR DESCRIPTION
**Pull request by jorisdral**

### CODE REVIEW

File: .github/workflows/haskell.yml [hunk 1]
```diff
@@ -45,7 +45,7 @@ jobs:
 
     - name: Configure the build
       run: |
-        cabal configure --enable-test --enable-benchmark --ghc-options="-Werror"
+        cabal configure --enable-test --enable-benchmark --ghc-options="-Werror" --ghc-options="-fno-ignore-asserts"
```
Comment (phadej) [inline .github/workflows/haskell.yml:48]: I never remember whether `--ghc-options` on command line affects all packages (i.e. also dependencies).

I'd rather do

```sh
cp cabal.project.ci.local cabal.project.local
```

with

```
package lsm-tree
  ghc-options: -Werror -fno-ignore-asserts
```

Comment (phadej) [reply .github/workflows/haskell.yml:48]: TL;DR

I advice to not ever use `cabal configure`. Or if you do, `cat cabal.project.local` afterwards.

Comment (jorisdral) [reply .github/workflows/haskell.yml:48]: https://cabal.readthedocs.io/en/stable/cabal-package.html#pkg-field-ghc-options doesn't specify, but https://cabal.readthedocs.io/en/stable/cabal-project.html#package-configuration-options does seem to provide some insight into the flag

> On the command line, these options are applied to all local packages. There is no per-package command line interface.

So I'm expecting it to apply only to local packages. I'm going to use `cat cabal.project.local` to verify

**Review by dcoutts — APPROVED** [review]

File: .github/workflows/haskell.yml (hunk 1 above)
Comment (jorisdral) [reply .github/workflows/haskell.yml:48]: Indeed, it only applies to local packages

**Pull request merged**